multus/multus.go

```diff
@@ -148,29 +148,53 @@ func delPlugins(exec invoke.Exec, argIfname string, delegates []*types.DelegateN
 	return nil
 }
 
-func cmdAdd(args *skel.CmdArgs, exec invoke.Exec, kubeClient k8s.KubeClient) (cnitypes.Result, error) {
-	var nopodnet bool
-	n, err := types.LoadNetConf(args.StdinData)
+// Attempts to load Kubernetes-defined delegates and add them to the Multus config.
+// Returns the number of Kubernetes-defined delegates added or an error.
+func tryLoadK8sDelegates(args *skel.CmdArgs, conf *types.NetConf, kubeClient k8s.KubeClient) (int, error) {
+	var err error
+
+	kubeClient, err = k8s.GetK8sClient(conf.Kubeconfig, kubeClient)
 	if err != nil {
-		return nil, fmt.Errorf("err in loading netconf: %v", err)
+		return 0, err
 	}
 
-	if n.Kubeconfig != "" {
-		delegates, err := k8s.GetK8sNetwork(args, n.Kubeconfig, kubeClient, n.ConfDir)
-		if err != nil {
-			if _, ok := err.(*k8s.NoK8sNetworkError); ok {
-				nopodnet = true
-			} else {
-				return nil, fmt.Errorf("Multus: Err in getting k8s network from pod: %v", err)
-			}
+	if kubeClient == nil {
+		if len(conf.Delegates) == 0 {
+			// No available kube client and no delegates, we can't do anything
+			return 0, fmt.Errorf("must have either Kubernetes config or delegates, refer Multus README.md for the usage guide")
 		}
+		return 0, nil
+	}
 
-		if err = n.AddDelegates(delegates); err != nil {
-			return nil, err
+	delegates, err := k8s.GetK8sNetwork(kubeClient, args, conf.ConfDir)
+	if err != nil {
+		if _, ok := err.(*k8s.NoK8sNetworkError); ok {
+			return 0, nil
+		} else {
```

Comment:
golint recognize that this 'else' is necessary and it makes Travis failure.
```
$ golint ./multus/... 
multus/multus.go:173:10: if block ends with a return statement, so drop this else and outdent its block (move short variable declaration to its own line if necessary)
```

Could you remove else and just return? I've checked it passes travis.